C/typeInference.c

```diff
@@ -649,9 +649,10 @@ type* mallocTypeInference(dag_node* dag, const size_t len, const combinator_coun
     , { .isBound = true, .bound = { .kind = PRODUCT, .arg = { &bound_var[5], &bound_var[5] } } }
     , { .isBound = true, .bound = { .kind = PRODUCT, .arg = { &bound_var[6], &bound_var[6] } } }
     , { .isBound = true, .bound = { .kind = PRODUCT, .arg = { &bound_var[7], &bound_var[7] } } }
+    , { .isBound = true, .bound = { .kind = PRODUCT, .arg = { &bound_var[8], &bound_var[8] } } }
     };
-  /* 'bound_var[8]' is bound to the type TWO^256. Eight non-trivial bindings were made. */
-  size_t bindings_used = 8;
+  /* 'bound_var[9]' is bound to the type TWO^256. Nine non-trivial bindings were made. */
```

Comment:
Is this supposed to say `TWO^512`, or is this change incorrect? I believe it is true that `bound_var[i]` is bound to type `TWO^i` for all `i`.

Reply:
For all `0 < i` we have that `bound_var[i]` is bound to the type `TWO^(2^(i-1))`.  FWIW, this all gets rewritten more clearly in
https://github.com/ElementsProject/simplicity/pull/11/commits/5430fae78b91561d9e66592ecf6858dde5b069e6

Reply:
After offline discussion I realized I was confusing `TWO^i` with `TWO^(2^i)`, which is a very different number